RiotSharp/FileCache.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.IO;
+using Newtonsoft.Json;
+
+namespace RiotSharp
+{
+    public class FileCache : ICache
+    {
+
+        /// <summary>
+        /// Create file cache instance
+        /// </summary>
+        /// <param name="dir">Directory for the cache to store in</param>
+        public FileCache(string dir = "/cache")
+        {
+            _directory = Path.Combine(Directory.GetCurrentDirectory(), dir);
+        }
+
+        private static string _directory;
+
+        private string GetPath(string key)
+        {
+            return Path.Combine(_directory, key.GetHashCode().ToString());
+        }
+
+        private T Load<T>(string path)
+        {
+            var json = File.ReadAllText(path);
+            return JsonConvert.DeserializeObject<T>(json);
+        }
+
+        private void Store<TK, TV>(TK key, TV value, long ttlMins = 24 * 60 * 7 * 4) // A month
+        {
+            CacheData<TV> data = new CacheData<TV>(ttlMins, value);
+            File.WriteAllText(GetPath(key.ToString()), JsonConvert.SerializeObject(data));
+        }
+
+        public bool IsExpired<T>(CacheData<T> data)
+        {
+            return data == null || DateTime.Now > data.CreatedAt.AddMinutes(data.TtlMinutes);
+        }
+
+        public void Add<TK, TV>(TK key, TV value, TimeSpan slidingExpiry) where TV : class
```

Comment:
could we add override to the methods inherited from `ICache`?

Reply:
So should I make it virtual void?

Reply:
public override void since it overrides a method in [ICache](https://github.com/BenFradet/RiotSharp/blob/afb5beabb52e95b29ac1ab103d84a04d08178ae7/RiotSharp/ICache.cs#L19)

Reply:
I don't think we have to do that.

Reply:
yeah but it makes it clearer that this method was inherited from ICache